central/scannerdefinitions/handler/handler_test.go

```diff
@@ -102,47 +100,42 @@ func (s *handlerTestSuite) postRequestV2() *http.Request {
 	return req
 }
 
-func (s *handlerTestSuite) postRequestV4() *http.Request {
-	// V4 ZIP file contents.
-	var v4Buf bytes.Buffer
-	zw := zip.NewWriter(&v4Buf)
-	file, err := zw.CreateHeader(&zip.FileHeader{
-		Name:               "manifest.json",
-		Comment:            "Scanner V4 manifest",
-		UncompressedSize64: uint64(len(v4ManifestContent)),
-	})
-	s.Require().NoError(err)
-	_, err = file.Write([]byte(v4ManifestContent))
-	s.Require().NoError(err)
-	s.Require().NoError(zw.Close())
+type zipBuilder struct {
+	buf *bytes.Buffer
+	zw  *zip.Writer
+}
 
+func newZipBuilder() *zipBuilder {
 	var buf bytes.Buffer
-	zw = zip.NewWriter(&buf)
-
-	// Currently, we need both V2 and V4 files when Scanner V4 is enabled.
-	file, err = zw.CreateHeader(&zip.FileHeader{
-		Name:               "scanner-defs.zip",
-		Comment:            "Scanner V2 content",
-		UncompressedSize64: uint64(len(content1)),
-	})
-	s.Require().NoError(err)
-	_, err = file.Write([]byte(content1))
-	s.Require().NoError(err)
+	return &zipBuilder{
+		buf: &buf,
+		zw:  zip.NewWriter(&buf),
+	}
+}
 
-	file, err = zw.CreateHeader(&zip.FileHeader{
-		Name:               "scanner-v4-defs.zip",
-		Comment:            "Scanner V4 content",
-		UncompressedSize64: uint64(v4Buf.Len()),
+func (b *zipBuilder) addFile(s *handlerTestSuite, name, comment string, content []byte) *zipBuilder {
+	s.Require().NotNil(b.buf)
+	file, err := b.zw.CreateHeader(&zip.FileHeader{
+		Name:               name,
+		Comment:            comment,
+		UncompressedSize64: uint64(len(content)),
 	})
 	s.Require().NoError(err)
-	_, err = file.Write(v4Buf.Bytes())
+	_, err = file.Write(content)
 	s.Require().NoError(err)
+	return b
+}
 
-	s.Require().NoError(zw.Close())
+func (b *zipBuilder) BuildBuffer(s *handlerTestSuite) *bytes.Buffer {
+	s.Require().NoError(b.zw.Close())
+	buf := b.buf
+	*b = zipBuilder{}
+	return buf
+}
 
-	req, err := http.NewRequestWithContext(s.ctx, http.MethodPost, "https://central.stackrox.svc/scannerdefinitions", &buf)
+func (s *handlerTestSuite) postRequestV4(buf *bytes.Buffer) *http.Request {
```

Comment:
the real requirement is here having `body io.Reader`. Since it's only used as an `io.Reader` can you update this to take in `body io.Reader`?

Reply:
7c786ab1ec